### DISCUSSION TITLE
Add Ldap Config During Setup by config or ENV variables

### DISCUSSION THREAD

**Discussion by compgeniuses** [zitadel/zitadel#8361] [Q&A]
SImilar to this here

```
curl -L 'https://$CUSTOM-DOMAIN/admin/v1/idps/ldap' \
-H 'Content-Type: application/json' \
-H 'Accept: application/json' \
-H 'Authorization: Bearer <TOKEN>' \
-d '{
  "name": "string",
  "servers": [
    "string"
  ],
  "startTls": true,
  "baseDn": "string",
  "bindDn": "string",
  "bindPassword": "string",
  "userBase": "string",
  "userObjectClasses": [
    "string"
  ],
  "userFilters": [
    "string"
  ],
  "timeout": "string",
  "attributes": {
    "idAttribute": "string",
    "firstNameAttribute": "string",
    "lastNameAttribute": "string",
    "displayNameAttribute": "string",
    "nickNameAttribute": "string",
    "preferredUsernameAttribute": "string",
    "emailAttribute": "string",
    "emailVerifiedAttribute": "string",
    "phoneAttribute": "string",
    "phoneVerifiedAttribute": "string",
    "preferredLanguageAttribute": "string",
    "avatarUrlAttribute": "string",
    "profileAttribute": "string"
  },
  "providerOptions": {
    "isLinkingAllowed": true,
    "isCreationAllowed": true,
    "isAutoCreation": true,
    "isAutoUpdate": true,
    "autoLinking": "AUTO_LINKING_OPTION_UNSPECIFIED"
  }
}'
```

IS it possible to have Ldap sync data automatically created During Initial Setup, while using DOcker compose?

**Comment by fforootd** [comment]
Zitadel ATM does not "sync" ldap but use it as an auth source and copy some data to its DB.

So what are you looking for in detail? To config LDAP through ENV or to have LDAP sync?

**Reply by compgeniuses** [reply]
to configure LDAP through ENV. that would really be helpful

**Reply by fforootd** [reply]
Got it, that would be a nice improvement.

@eliobischof are you aware if we have an issue for config by file already?

**Reply by eliobischof** [reply]
> Got it, that would be a nice improvement.
> 
> @eliobischof are you aware if we have an issue for config by file already?

We don't support setting up IDPs via runtime config ATM. This needs to be done in a subsequent job via API.

**Reply by compgeniuses** [reply]
IF this is possible via API, IS it also not possible to generate and acquire the initial API keys required to handle this job Automatically.

The case to be solved is, One installs Zitadel, selects the AD or Ldap Instance, the IDP is pre-configured, then the user can login to the platform to handle additional details.

If this is achievebale, thn it solves my usecase 100%

**Reply by fforootd** [reply]
You can deploy a zitadel with a service account https://zitadel.com/docs/self-hosting/deploy/compose#docker-compose-with-service-account and then create the resources by either calling the API or use Terraform to provision them.

**Reply by compgeniuses** [reply]
am i missing anything here 
# then you can move your machine key
mv ./machinekey/zitadel-admin-sa.json $HOME/zitadel-admin-sa.json

I am more lost now, its looking like the tutorial is not complete

**Reply by fforootd** [reply]
If you start the docker-compose example it should output a file to your host at the directory mentioned in the mv command.

Check here https://github.com/zitadel/zitadel/blob/main/docs/docs/self-hosting/deploy/docker-compose-sa.yaml#L31
